core/Pos/Binary/Core/Types.hs

```diff
@@ -38,17 +35,16 @@ instance Bi T.CoinPortion where
     get = label "CoinPortion" $ get >>= T.mkCoinPortion
 
 instance Bi T.LocalSlotIndex where
-    get = label "LocalSlotIndex" $ T.LocalSlotIndex . getUnsignedVarInt <$> get
-    put (T.LocalSlotIndex c) = put (UnsignedVarInt c)
+    get =
+        label "LocalSlotIndex" $
+        (eitherToFail . T.mkLocalSlotIndex) . getUnsignedVarInt =<< get
```

Comment:
Not needed, but it's insignificant. Maybe I will fix, maybe someone else will do it.